RiotSwiftUI/Modules/Room/PollTimeline/Coordinator/PollTimelineCoordinator.swift

```diff
@@ -0,0 +1,143 @@
+
+    // Must be used only internally
+    var childCoordinators: [Coordinator] = []
+    
+    // MARK: - Setup
+    
+    @available(iOS 14.0, *)
+    init(parameters: PollTimelineCoordinatorParameters) throws {
+        self.parameters = parameters
+        
+        try pollAggregator = PollAggregator(session: parameters.session, room: parameters.room, pollStartEvent: parameters.pollStartEvent)
+        pollAggregator.delegate = self
+        
+        pollTimelineViewModel = PollTimelineViewModel(timelinePoll: mapPoll(pollAggregator.poll))
+        pollTimelineViewModel.callback = { [weak self] result in
+            guard let self = self else { return }
+            
+            switch result {
+            case.selectedAnswerOptionsWithIdentifiers(let identifiers):
+                self.selectedAnswerIdentifiersSubject.send(identifiers)
+            }
+        }
+        
+        selectedAnswerIdentifiersSubject
+            .debounce(for: 1.0, scheduler: RunLoop.main)
+            .removeDuplicates()
+            .sink { [weak self] identifiers in
+                guard let self = self else { return }
+
+                self.parameters.room.sendPollResponse(for: parameters.pollStartEvent,
+                                                      withAnswerIdentifiers: identifiers,
+                                                      localEcho: nil, success: nil) { [weak self] error in
+                    guard let self = self else { return }
+                    
+                    MXLog.error("[PollTimelineCoordinator]] Failed submitting response with error \(String(describing: error))")
+                    
+                    self.pollTimelineViewModel.dispatch(action: .showAnsweringFailure)
+                }
+            }
+            .store(in: &cancellables)
+    }
+    
+    // MARK: - Public
+    func start() {
+        
+    }
+    
+    func toPresentable() -> UIViewController {
+        return VectorHostingController(rootView: PollTimeline(viewModel: pollTimelineViewModel.context))
+    }
+    
+    func canEndPoll() -> Bool {
+        return (pollAggregator.poll.isClosed == false)
+    }
+    
+    func endPoll() {
+        self.parameters.room.sendPollEnd(for: parameters.pollStartEvent, localEcho: nil, success: nil) { error in
+            self.pollTimelineViewModel.dispatch(action: .showClosingFailure)
+        }
+    }
+    
+    // MARK: - PollAggregatorDelegate
+    
+    func pollAggregatorDidUpdateData(_ aggregator: PollAggregator) {
+        pollTimelineViewModel.dispatch(action: .updateWithPoll(mapPoll(aggregator.poll)))
+    }
+    
+    func pollAggregatorDidStartLoading(_ aggregator: PollAggregator) {
+        
+    }
+    
+    func pollAggregatorDidEndLoading(_ aggregator: PollAggregator) {
+        
+    }
+    
+    func pollAggregator(_ aggregator: PollAggregator, didFailWithError: Error) {
+        
+    }
+    
+    // MARK: - Private
+    
+    func mapPoll(_ poll: PollProtocol) -> TimelinePoll {
```

Comment:
I had to look at this to know what it does. Maybe `makeTimelinePoll(from:)` or add a doc comment? Or both!
**Edit:** Or tbh, if it's an option, perhaps add a convenience `init(poll: PollProtocol)` to `TimelinePoll`?